theano/configdefaults.py

```diff
@@ -1524,11 +1536,24 @@ def _timeout_default():
 
 try:
     p_out = output_subprocess_Popen([config.cxx, '-dumpversion'])
+    cxx_version_str = p_out[0].strip().decode()
+except OSError:
+    # Typically means gcc cannot be found.
+    cxx_version_str = 'CXX_NOT_FOUND'
+
+try:
+    p_out = output_subprocess_Popen(['g++', '-dumpversion'])
     gcc_version_str = p_out[0].strip().decode()
 except OSError:
     # Typically means gcc cannot be found.
-    gcc_version_str = 'GCC_NOT_FOUND'
+    gcc_version_str = 'CXX_NOT_FOUND'
 
+try:
+    p_out = output_subprocess_Popen([config.cxx, '-dumpversion'])
+    icc_version_str = p_out[0].strip().decode()
+except OSError:
+    # Typically means gcc cannot be found.
+    icc_version_str = 'CXX_NOT_FOUND'
```

Comment:
This is the same code as for cxx_version_str. What about having only one cxx_version_str and remove icc_version_str and gcc_version_str and prepends to it config.cxx?

It would change the semantic to not only a version number. We could also do that without the prepend. What do you think? @lamblin @abergeron?